src/usr/local/www/firewall_rules_edit.php

```diff
@@ -295,6 +317,10 @@ function is_aoadv_used($rule_config) {
 
 	unset($input_errors);
 
+	if (!array_key_exists($_POST['ipprotocol'], $icmplookup)) {
+		$input_errors[] = gettext("The IP protocol is not recognized."); 
+	}
+
```

Comment:
Fixed I think? Can you check as it's hard to be sure, the extraneous chars don't show up in the GitHub UI